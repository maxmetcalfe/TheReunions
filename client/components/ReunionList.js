import React, { Component } from "react";
import PropTypes from "prop-types";
import constants from "../constants";


class ReunionList extends Component {

  render() {

    if (!this.props.reunions) {
      return null;
    }

    return (
      <div id="reunion-list">
        <table>
          <tbody>
            {this.props.reunions.features.map((reunion, index) => (
              <tr className={constants.CATEGORIES[reunion.properties.category] || constants.CATEGORIES["Cat 4"]} key={index}>
                <td key={index}>{reunion.properties.location} - {reunion.properties.name}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

ReunionList.propTypes = {
  reunions: PropTypes.object
};

export default ReunionList;
