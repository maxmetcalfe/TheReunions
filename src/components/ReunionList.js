import React, { Component } from "react";

class ReunionList extends Component {

  render() {
    console.log(this.props);
    return (
      <div id="reunion-list">
        <table>
          <tbody>
            {this.props.reunions.features.map((reunion) => (
              <tr key={reunion.properties.element_id} className={"row-" + reunion.properties.category}>
                <td>{reunion.properties.name}</td>
                <td>{reunion.properties.location}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

export default ReunionList;
