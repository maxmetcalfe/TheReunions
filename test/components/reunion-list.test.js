import React from "react"
import renderer from "react-test-renderer"
import ReunionList from "../../client/components/ReunionList"

test("ReunionList", () => {
  let reunions = {
    features: [
      {
        properties: {
          name: "Boston 2014",
          location: "Boston",
          category: "Cat 4"
        }
      },
      {
        properties: {
          name: "Charleston 2012",
          location: "Charleston",
          category: "Cat 3"
        }
      },
      {
        properties: {
          name: "New York 2011",
          location: "New York",
          category: "Cat 2"
        }
      }
    ]
  }

  let component = renderer.create(
    <ReunionList reunions={reunions} ></ReunionList>
  )

  expect(component).toMatchSnapshot()
})

test("ReunionList, single reunion", () => {
  let reunions = {
    features: [
      {
        properties: {
          name: "Boston 2014",
          location: "Boston",
          category: "Cat 4"
        }
      }
    ]
  }

  let component = renderer.create(
    <ReunionList reunions={reunions} ></ReunionList>
  )

  expect(component).toMatchSnapshot()
})

test("ReunionList, null reunions", () => {

  let component = renderer.create(
    <ReunionList ></ReunionList>
  )

  expect(component).toMatchSnapshot()
})