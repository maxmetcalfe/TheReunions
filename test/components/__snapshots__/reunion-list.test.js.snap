// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReunionList 1`] = `
<div
  id="reunion-list"
>
  <table>
    <tbody>
      <tr
        className="two"
      >
        <td>
          New York
           - 
          New York 2011
        </td>
      </tr>
      <tr
        className="three"
      >
        <td>
          Charleston
           - 
          Charleston 2012
        </td>
      </tr>
      <tr
        className="four"
      >
        <td>
          Boston
           - 
          Boston 2014
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ReunionList, single reunion 1`] = `
<div
  id="reunion-list"
>
  <table>
    <tbody>
      <tr
        className="Cat 4"
      >
        <td>
          Boston
           - 
          Boston 2014
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
